src/map/entities/charentity.cpp

```diff
@@ -1587,6 +1596,13 @@ void CCharEntity::OnRaise()
             hpReturned             = (uint16)GetMaxHP();
             ratioReturned          = ((GetMLevel() <= 50) ? 0.50f : 0.90f) * (1 - map_config.exp_retain);
         }
+        else if (m_hasRaise == 5)
```

Comment:
seems t4 and t4 have identical code here, but I can't locate where in the project we actually set t4

Reply:
yeah that baffled me also, but i did not want to touch it as it was obviously put there for a reason

Reply:
unless this was for arise but never commented as such

Reply:
after trying to dig around in the project..I think that may be the case. Gonna see what git blame says it came in with

Reply:
@claywar git blame points its finger at you, was this intended for Arise ?
https://github.com/LandSandBoat/server/commit/053211ee5af7db48830776adaee9ec4a7ee7aab1#diff-81491448fbf9d158d4cc9e7df3822956203b840795b53b030ee1be7f250dd7cfR1575

Reply:
Ah, Clay added it for Reraise 4 from the job point additions

Reply:
There is a reraise 4?  /noob

Reply:
from BG
Grants an [Arise](https://www.bg-wiki.com/ffxi/Arise) effect upon KO that restores 100% HP and 90% XP lost and grants 3 minute weakness.

Reply:
so yup, can use it

Reply:
I have to keep manually refreshing to see your messages, github being garbage today

Reply:
yeah same

Reply:
ok, ill change this to use 4 instead, and that means reraise 4 can be plugged it also

Reply:
I believe that was meant for Arise, so yes.

Reply:
I guess Arise is basically raise4 to match that reraise4, but with the added benefit of working for the dead person _twice_ (since your whm isn't dead on self casting, they only get the reraise half of it)

Reply:
yeah, reraise with reraise

Reply:
Yo dawg, I heard you like reraise...